app/models/form_profile.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+# TODO(AJD): Virtus POROs for now, will become ActiveRecord when the profile is persisted
+class FormFullName
+  include Virtus.model
+
+  attribute :first, String
+  attribute :middle, String
+  attribute :last, String
+  attribute :suffix, String
+end
+
+class FormAddress
+  include Virtus.model
+
+  attribute :street
+  attribute :street_2
+  attribute :city
+  attribute :state
+  attribute :country
+  attribute :postal_code
+end
+
+class FormApplicantInformation
+  include Virtus.model
+
+  attribute :full_name, FormFullName
+  attribute :date_of_birth, Date
+  attribute :gender, String
+end
+
+class FormContactInformation
+  include Virtus.model
+
+  attribute :address, FormAddress
+  attribute :home_phone, String
+end
+
+class FormProfile
+  include Virtus.model
+
+  attribute :applicant_information, FormApplicantInformation
+  attribute :contact_information, FormContactInformation
+
+  # Collects data the VA has on hand for a user. The data may come from many databases/services.
+  # In case of collisions, preference is given in this order:
+  # * The form profile cache (the record for this class)
+  # * ID.me
+  # * MVI
+  # * TODO(AJD): MIS (military history)
+  #
+  def prefill_form(form_id, user)
+    @applicant_information = initialize_application_information(user)
+    @contact_information = initialize_contact_information(user)
+    mappings = mappings_for_form(form_id)
+    generate_prefill(mappings)
+  end
+
+  private
+
+  def initialize_application_information(user)
+    FormApplicantInformation.new(
+      full_name: {
+        first: user.first_name&.capitalize,
+        middle: user.middle_name&.capitalize,
+        last: user.last_name&.capitalize,
+        suffix: user.va_profile[:suffix]
+      },
+      date_of_birth: user.birth_date,
+      gender: user.gender
+    )
+  end
+
+  def initialize_contact_information(user)
+    FormContactInformation.new(
+      address: {
+        street: user.va_profile[:address][:street_address_line],
+        street2: nil,
+        city: user.va_profile[:address][:city],
+        state: user.va_profile[:address][:state],
+        postal_code: user.va_profile[:address][:postal_code],
+        country: user.va_profile[:address][:country]
+      },
+      home_phone: user.va_profile[:home_phone]
+    )
+  end
+
+  def mappings_for_form(form_id)
+    @mappings ||= Hash.new do |h, key|
```

Comment:
Right, but won't `mappings_for_form` only once per request via a `FormProfile.new`, giving a new `@mappings` each time? I may be missing something, but, I'm not spotting the gain from using the hash at all rather than just returning the contents of the `YAML.load`.